doc/stdenv/stdenv.xml

```diff
@@ -2111,16 +2111,106 @@ postInstall = ''
       </para>
      </listitem>
     </varlistentry>
+
     <varlistentry>
      <term>
-      cmake
+      <anchor xml:id="cmake"/>
+       cmake
      </term>
      <listitem>
       <para>
-       Overrides the default configure phase to run the CMake command. By default, we use the Make generator of CMake. In addition, dependencies are added automatically to CMAKE_PREFIX_PATH so that packages are correctly detected by CMake. Some additional flags are passed in to give similar behavior to configure-based packages. You can disable this hook’s behavior by setting configurePhase to a custom value, or by setting dontUseCmakeConfigure. cmakeFlags controls flags passed only to CMake. By default, parallel building is enabled as CMake supports parallel building almost everywhere. When Ninja is also in use, CMake will detect that and use the ninja generator.
+       Overrides the default configure phase to run the CMake command. By default, we use the Make generator of CMake, however, if ninja is present, then the ninja generator will be used instead. In addition, dependencies are added automatically to CMAKE_PREFIX_PATH so that packages are correctly detected by CMake. By default, the cmake configurePhase can be thought of as:
+       <programlisting>
+mkdir build
+cd build
+cmake ..</programlisting>
       </para>
+      <variablelist>
+       <title>Variables controlling CMake</title>
+       <varlistentry>
+        <term>
+         <varname>cmakeFlags</varname>
```

Comment:
It might be also nice to add anchors to the individual variables – I did that for most of stdenv in 3f2b7beaf337818ea77014c6d89d84926b4ed463 but somehow missed Meson, which was annoying when I wanted to link it recently.